llm/server.go

```diff
@@ -474,7 +474,16 @@ func (s *llamaServer) Load(ctx context.Context, systemInfo ml.SystemInfo, system
 		s.mem.GPUs[i].Cache = make([]uint64, s.totalLayers)
 	}
 
-	kv, graphPartialOffload, graphFullOffload := s.ggml.GraphSize(uint64(s.options.NumCtx), uint64(s.loadRequest.BatchSize),
+	// Check if embedding model and adjust batch size accordingly
+	batchSize := s.loadRequest.BatchSize
+	_, isEmbedding := s.ggml.KV()[fmt.Sprintf("%s.pooling_type", s.ggml.KV().Architecture())]
+	if isEmbedding {
+		batchSize = s.options.NumCtx
+		s.loadRequest.BatchSize = batchSize
```

Comment:
Good catch that we weren't actually setting `loadRequest`. It probably would be better to just operate on that rather than introducing a new variable to avoid potential issues like this.

You could also check that the batch size is less than the context to avoid logging when the batch size is already sufficient, similar to the Ollama engine.